samples/GalleryApp/MainPage.xaml.cs

```diff
@@ -21,29 +16,81 @@ namespace GalleryApp
     /// </summary>
     public sealed partial class MainPage : Page
     {
+        public ObservableCollection<SkillView> SkillPages { get; } = new ObservableCollection<SkillView>();
+
+        private static string m_filePath = "\\Pages\\SkillViewGlossary.json";
+
+        private static List<SkillCategory> m_allCategories;
+        private static SemaphoreSlim m_semaphore = new SemaphoreSlim(1);
```

Comment:
No need of using threads here because there's only one task inside LoadAllSkills.

---
In reply to: [302170202](https://github.com/microsoft/WindowsVisionSkillsPreview/pull/45#discussion_r302170202) [](ancestors = 302170202)